### ISSUE TITLE
Check out this one simple trick to absolutely eat someone's storage with a single image file

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#5879]
**Describe the bug**
The game currently writes errors encountered when loading the game to an error log. That's all fine and dandy, but it's currently possible to cause this error log to balloon in size with the inclusion of a single image file to the game.

**To Reproduce**
1. Rename or create an image file with a name that denotes a very high frame number. i.e. (<name>~X.png) where X is a very large number.
2. Launch the game.
3. Observe how your error file gets filled with line upon line of `Sprite "<path>": missing frame X.`
4. If the image you chose also has a single 2x version present, you'll also get `Sprite "<path>": missing @2x frame X` for every frame, doubling the number of errors.

**Expected behavior**
Under no circumstances should a single file be capable of filling the error log with so much junk.

**Screenshots**
After renaming `typhoon~3.png` to `typhoon~3000000.png` and launching the game, my error log grew to nearly a third of a gig in size. Notepad++ can't even function opening a text file of that size. Two more zeros and I would have eaten all the remaining space on my SSD and then some; not even sure what would happen after the game ran out of memory to write to.
![image](https://user-images.githubusercontent.com/17688683/115754675-4b8ad400-a351-11eb-9f23-ffea3bfea445.png)

**System (please complete the following information):**
 - OS: Windows 10.
 - Game Source: Self-compiled.
 - Version: Latest commit as of posting (e19c8e2).

**Link to save file**
I'd link the error file to be funny, but GitHub has a 10mb upload limit.

**Additional context**
The following code is the source of all the errors. If an image file is named with a very high frame number, the game expects all frames up to that number to exist, and throws an error for each and every one. 
https://github.com/endless-sky/endless-sky/blob/576f7a72678ae0fafe8a24c9494f34c62f51c847/source/ImageSet.cpp#L171-L186

Suggested that the error log instead list the *number* of frames missing instead of individually listing each frame that is missing.

**Comment by Amazinite** [comment]
This is like an ASAP type of thing to fix to me. Not that I expect anyone to maliciously use this, but it's certainly a possibility, and I only caught it because someone for some reason had a 3mb error log filled with "missing frame" errors for thousands of lines, so clearly at least one plugin has run into this before, and to be honest this isn't the type of thing that I expect you to accidentally cause to happen without immediately fixing it if you noticed it.

**Comment by MCOfficer** [comment]
The reasonable thing is to report this in ranges, i.e.
```
Sprite "<path>": missing frames X-Y.
Sprite "<path>": missing @2x frames X-Y.
```

This would still allow intentional abuse by alternating between existent and nonexistent images (`typhoon~1.png`, `typhoon~3.png`, `typhoon~5.png`), but that would require one image file for every produced log entry - at that point your main concern isn't the storage used by the *log file*.

**Comment by Amazinite** [comment]
I don't see ranges as necessary, as in terms of normal use you're probably only using up to double digit frames anyway, so if the game reports that 3 frames are missing, it shouldn't be too difficult to track them down yourself.
I don't see implementing reporting ranges as much more difficult than simply reporting the number of missing frames, though.

**Comment by ashdnazg** [comment]
A good compromise is to write the number of missing frames and the first range, then it's both useful and bound

**Comment by Zitchas** [comment]
Can it count instances before writing them? Thus it could print out an arbitrary number of missing frames, or ranges where possible, but if the number of missing frames exceeds a fixed limit, it simply says something to the effect of 
`Sprite "<path>": missing X frames.`

**Comment by InfiniteDonuts** [comment]
I wrote what I _think_ is a solution, but I haven't had time to test it yet

**Comment by Amazinite** [comment]
> Just out of curiosity, how long did it take to write a 300Mb log file?

About 30 seconds, and then the game doesn't properly launch sometimes.

**Comment by ashdnazg** [comment]
@InfiniteDonuts oi, sorry, I missed the fact you were already working on this :S

**Comment by InfiniteDonuts** [comment]
@ashdnazg Totally OK dude, I was just bored :)

**Comment by InfiniteDonuts** [comment]
I was going to wait until I got home to test it as I was using a Gitpod workspace (and ES doesn't take kindly to that apparently), and I also asked the folks on the Discord to review.

Also, looking back on my commit, it looks like I forgot to re-rename the electron beam sprite back to the default... my bad

**Comment by tehhowch** [comment]
Fixed by #5881
